tests/validator_stakes/validator_stakes_test.go

```diff
@@ -0,0 +1,157 @@
+				increaseValidatorStake(t, net, sfcContract, id, utils.ToFtm(625))
+			}
+			net.AdvanceEpoch(t, 1)
+
+			// Test new stake distribution
+			validators, secondEpoch := getValidatorsInCurrentEpoch(t, sfcContract)
+			require.Condition(t, func() bool {
+				next := new(big.Int).Add(firstEpoch, big.NewInt(1))
+				return secondEpoch.Cmp(next) == 0
+			}, "epoch did not advance as expected")
+			requireValidatorStakesEqualTo(t, validators, []uint64{750, 750, 750, 750, 750, 750})
+		})
+
+	t.Run("third epoch is dominated by two validators again",
+		func(t *testing.T) {
+			// Prepare new stake distribution
+			for _, id := range []idx.ValidatorID{5, 6} {
+				increaseValidatorStake(t, net, sfcContract, id, utils.ToFtm(1_000_000))
+			}
+			net.AdvanceEpoch(t, 1)
+
+			// Test new stake distribution
+			validators, secondEpoch := getValidatorsInCurrentEpoch(t, sfcContract)
+			require.Condition(t, func() bool {
+				next := new(big.Int).Add(firstEpoch, big.NewInt(2))
+				return secondEpoch.Cmp(next) == 0
+			}, "epoch did not advance as expected")
+			requireValidatorStakesEqualTo(t, validators, []uint64{750, 750, 750, 750, 1000750, 1000750})
+		})
+}
+
+func increaseValidatorStake(t *testing.T, net *tests.IntegrationTestNet, sfcContract *sfc100.Contract, id idx.ValidatorID, amount *big.Int) {
+	t.Helper()
+
+	// First endow the validator account with enough FTM to stake.
+	// If this step is not done, the account calling sfc.Delegate will be
+	// delegating into the stake. The interaction with the sfc is simpler
+	// if there is only one delegator, which is the validator account itself.
+	account := tests.Account{
+		PrivateKey: makefakegenesis.FakeKey(id),
+	}
+	receipt, err := net.EndowAccount(account.Address(), amount)
+	require.NoError(t, err)
+	require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status)
+
+	// Then the valdator account delegates stake to its validatorID.
+	opts, err := net.GetTransactOptions(&account)
+	require.NoError(t, err)
+	opts.Value = amount
+	tx, err := sfcContract.Delegate(opts, big.NewInt(int64(id)))
+	require.NoError(t, err)
+
+	receipt, err = net.GetReceipt(tx.Hash())
+	require.NoError(t, err)
+	require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status)
+}
+
+// requireValidatorStakesEqualTo checks that the stakes of the validators
+// in the 'validators' map match the expected stakes provided in 'expectedStakes' slice.
+// The index of the expectedStakes slice corresponds to the ValidatorID - 1.
+func requireValidatorStakesEqualTo(t *testing.T, validators map[idx.ValidatorID]*big.Int, expectedStakes []uint64) {
+	t.Helper()
+
+	for id, expectedStake := range expectedStakes {
+		stake, ok := validators[idx.ValidatorID(id+1)]
+		require.True(t, ok, "validator %d not found", id+1)
+		expectedBig := utils.ToFtm(expectedStake)
+		require.Equal(t, 0, stake.Cmp(expectedBig),
+			"validator %d stake mismatch: expected %s, got %s", id+1, expectedBig.String(), stake.String())
```

Comment:
```suggestion
		require.Equal(t, expectedBig.Uint64(), stake().Uint64(),
			"validator %d stake mismatch: expected %s, got %s", id+1, expectedBig.String(), stake.String())
```
easier to read than interpreting result of cmp. wdyt ?